src/ledger/LedgerManagerImpl.cpp

```diff
@@ -582,6 +595,7 @@ LedgerManagerImpl::closeLedger(LedgerCloseData const& ledgerData)
     std::unique_ptr<LedgerCloseMeta> ledgerCloseMeta;
     if (mMetaStream)
     {
+        emitAnyStoredSafeMetaAfterLedgerValidation();
```

Comment:
should there be some invariant check here that the meta that is emitted corresponds to what we check?

Reply:
Yeah, that'd be good -- and I think that this suggestion of yours would give us the parameter that would allow us to do that, as well as making the interface harder to misuse:

> It looks to me that the way to "idiot proof" this is to instead make it that there is a variant of LedgerManagerImpl::closeLedger that exists that takes the expected lcl hash as a parameter, and if present (and matches) we can emit meta right away (sharing the code path with the UNSAFE_EMIT_META==true).

Reply:
I've added a `releaseAssert` there.